plugin/pkg/admission/runtimeclass/admission.go

```diff
@@ -160,25 +160,24 @@ func (r *RuntimeClass) getRuntimeClass(pod *api.Pod, runtimeClassName *string) (
 
 func setOverhead(a admission.Attributes, pod *api.Pod, runtimeClass *v1beta1.RuntimeClass) (err error) {
 
-	if runtimeClass != nil {
-		if runtimeClass.Overhead != nil {
-
-			// convert to internal type and assign to pod's Overhead
-			nodeOverhead := &node.Overhead{}
-			err := nodev1beta1.Convert_v1beta1_Overhead_To_node_Overhead(runtimeClass.Overhead, nodeOverhead, nil)
-			if err != nil {
-				return err
-			}
+	if runtimeClass == nil || runtimeClass.Overhead == nil {
+		return nil
+	}
 
-			// reject pod if Overhead is already set that differs from what is defined in RuntimeClass
-			if pod.Spec.Overhead != nil && !apiequality.Semantic.DeepEqual(nodeOverhead.PodFixed, pod.Spec.Overhead) {
-				return admission.NewForbidden(a, fmt.Errorf("pod rejected: Pod's Overhead doesn't match RuntimeClass's defined Overhead"))
-			}
+	// convert to internal type and assign to pod's Overhead
+	nodeOverhead := &node.Overhead{}
+	err := nodev1beta1.Convert_v1beta1_Overhead_To_node_Overhead(runtimeClass.Overhead, nodeOverhead, nil)
```

Comment:
Because `err` is a named return value, we don't need `:=`. Alternatively, we can not use a named return value, which I think most likely makes more sense.

Reply:
Good catch.  Updated.